lib/DB.php

```diff
@@ -240,6 +240,28 @@ public function tableExists($sTableName)
         return ($this->getOne($sSQL, array(':tablename' => $sTableName)) == 1);
     }
 
+    /**
+    * Returns a list of table names in the database
+    *
+    * @return array[]
+    */
+    public function getListOfTables()
+    {
+        return $this->getCol("SELECT tablename FROM pg_tables WHERE schemaname='public'");
+    }
+
+    /**
+     * Deletes a table. Returns true on success. Returns true if the table didn't exist.
```

Comment:
If the table didn't exist in the first place that's treated as the deletion succeeding (because your goal was for the database to be deleted). I reworded it now.